lib/logstash/inputs/rabbitmq.rb

```diff
@@ -93,15 +97,38 @@ def declare_queue
       def consume!
         # we manually build a consumer here to be able to keep a reference to it
         # in an @ivar even though we use a blocking version of HB::Queue#subscribe
-        @consumer = @hare_info.queue.build_consumer(:block => true) do |metadata, data|
-          @codec.decode(data) do |event|
-            decorate(event)
-            @output_queue << event if event
+
+        # The logic here around resubscription might seem strange, but its predicated on the fact
+        # that we rely on MarchHare to do the reconnection for us with auto_reconnect.
+        # Unfortunately, while MarchHare does the reconnection work it won't re-subscribe the consumer
+        # hence the logic below.
+        loop do
+          begin
+            @consumer = @hare_info.queue.build_consumer(:block => true) do |metadata, data|
+              @codec.decode(data) do |event|
+                decorate(event)
+                @output_queue << event if event
+              end
+              @hare_info.channel.ack(metadata.delivery_tag) if @ack
+            end
+
+            if @hare_info.connection.connected?
+              @logger.info("Will subscribe with consumer", :config => config)
+              @hare_info.queue.subscribe_with(@consumer, :manual_ack => @ack, :block => true)
+              @logger.warn("Queue subscription ended! Will retry in #{@subscription_retry_interval_seconds}s", :config => config)
+            else
+              @logger.info("RabbitMQ connection down, will wait to retry subscription", :config => config)
+            end
+
+            break if stop?
+            @consumer.gracefully_shut_down # Try to clean up gracefully
+          rescue MarchHare::Exception => e
```

Comment:
hmmm..Is there a specific exception that gets thrown? or a blanket `MarchHare::Exception`